### PR TITLE
CORE-17369: Enable display of version information in CLI plugins

### PR DESCRIPTION
**Pull request by josephzunigadaly**
Example output:

```text
generatedScripts $ ./corda-cli.sh database --version
db-config 5.2.0.0-SNAPSHOT

Provider: R3 Ltd

Commit ID: edeca54251cc7c7e088d123c1aa913c31e7a8145

generatedScripts $ ./corda-cli.sh initial-config --version
initial-config 5.2.0.0-SNAPSHOT

Provider: R3 Ltd

Commit ID: edeca54251cc7c7e088d123c1aa913c31e7a8145

generatedScripts $ ./corda-cli.sh initial-rbac --version
initial-rbac 5.2.0.0-SNAPSHOT

Provider: R3 Ltd

Commit ID: edeca54251cc7c7e088d123c1aa913c31e7a8145

generatedScripts $ ./corda-cli.sh network --version
network 5.2.0.0-SNAPSHOT

Provider: R3 Ltd

Commit ID: edeca54251cc7c7e088d123c1aa913c31e7a8145

generatedScripts $ ./corda-cli.sh mgm --version
network 5.2.0.0-SNAPSHOT

Provider: R3 Ltd

Commit ID: edeca54251cc7c7e088d123c1aa913c31e7a8145

generatedScripts $ ./corda-cli.sh package --version
package 5.2.0.0-SNAPSHOT

Provider: R3 Ltd

Commit ID: edeca54251cc7c7e088d123c1aa913c31e7a8145

generatedScripts $ ./corda-cli.sh secret-config --version
secret-config 5.2.0.0-SNAPSHOT

Provider: R3 Ltd

Commit ID: edeca54251cc7c7e088d123c1aa913c31e7a8145

generatedScripts $ ./corda-cli.sh topic --version
topic-config 5.2.0.0-SNAPSHOT

Provider: R3 Ltd

Commit ID: edeca54251cc7c7e088d123c1aa913c31e7a8145

generatedScripts $ ./corda-cli.sh vnode --version
virtual-node 5.2.0.0-SNAPSHOT

Provider: R3 Ltd

Commit ID: edeca54251cc7c7e088d123c1aa913c31e7a8145

generatedScripts $ ./corda-cli.sh preinstall --version
preinstall 5.2.0.0-SNAPSHOT

Provider: R3 Ltd

Commit ID: edeca54251cc7c7e088d123c1aa913c31e7a8145

```

### CODE REVIEW

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Review by kyriathar — APPROVED** [review]
OK (db & packaging)

**Review by ronanbrowne — APPROVED** [review]
LGTM (blt files)

File: tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/NetworkPluginWrapper.kt [hunk 1]
```diff
@@ -44,7 +48,8 @@ class NetworkPluginWrapper : Plugin() {
             GenerateGroupPolicy::class,
         ],
         mixinStandardHelpOptions = true,
-        description = ["Plugin for membership operations."]
+        description = ["Plugin for membership operations."],
+        versionProvider = VersionProvider::class
```
Comment (yift-r3) [inline tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/NetworkPluginWrapper.kt:52]: Adding a comma after the last argument will reduce the change in the next commit...

**Review by yift-r3 — APPROVED** [review]
Can we add the build date as well?

File: tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/NetworkPluginWrapper.kt (hunk 1 above)
Comment (josephzunigadaly) [reply tools/plugins/network/src/main/kotlin/net/corda/cli/plugins/network/NetworkPluginWrapper.kt:52]: Thank you for the feedback Yiftach. Adding a build date would be a decision against doing reproducible builds. So it's a bigger thing than just changing the output.

**Pull request merged**